modules/core/packer/roles/nomad/files/configure.sh

```diff
@@ -202,13 +204,97 @@ EOF
   chown "${user}:${user}" "${config_dir}/acl.hcl"
 }
 
+function generate_docker_config {
+  local readonly consul_prefix="${1}"
+  local readonly config_dir="${2}"
+  local readonly user="${3}"
+  local readonly consul_template_config="${4}"
+  local readonly docker_auth_path="${5}"
+
+    local docker_config=$(cat <<EOF
+client {
+  options {
+    "docker.auth.config" = "${docker_auth_path}"
+  }
+}
+EOF
+)
+
+  log_info "Writing Docker configuration to ${config_dir}/docker.hcl"
+  echo "${docker_config}" > "${config_dir}/docker.hcl"
+  chown "${user}:${user}" "${config_dir}/docker.hcl"
+
+  local vault_path
+  vault_path=$(consul_kv "${consul_prefix}docker-auth/path")
+
+  local docker_template=$(cat <<EOF
+template {
+  destination = "${docker_auth_path}"
+  create_dest_dirs = true
+
+  # consul-template does not deal with ownership properly
+  # See https://github.com/hashicorp/consul-template/issues/1061
+  command = "bash -c 'chown ${user}:${user} ${docker_auth_path}'"
+
+  perms = 0600
+  error_on_missing_key = true
+
+  # The goal is to produce something like
+  # {
+  #     "auths": {
+  #         "registry.a.b": {
+  #           "auth": "aaaaa="
+  #         },
+  #         "foo.bar.xyz": {
+  #           "auth": "bbbb="
+  #         }
+  #     }
+  # }
+  #
+  # But two things make it difficult:
+  # 1. JSON doesn't do dangling commas
+  # 2. Go Template has really basic logic handling
+  #
+  # Thus this template is more complicated than needed.
+  contents = <<EOH
+{{- define "keys" -}}
+  {{- with secret "${vault_path}" }}
+      {{- range \$key, \$value := .Data -}}
+        {{ \$key }}{{ " " }}
```

Comment:
I suppose this is to flatten Object into an Array? Why is `{{ " " }}` required here though?

Reply:
I am trying to join the list of keys. I could use `comma` as a separator, but later on when I use `spllit` by comma, the dangling comma will create a final empty item. There is no easy way to substring. BUT there is a `trimSpace` function. So I just use space as the separator so that the dangling space can be trimmed off.

Go Template **is horrible**.

Reply:
Would be nice if they had something like a Python string `join` haha

Reply:
There is a join. But I can't get a list of map keys only. LOL